revdep/problems.md

```diff
@@ -1,3 +1,217 @@
+
+* Version: 1.3.15
+* GitHub: https://github.com/santikka/causaleffect
+* Source code: https://github.com/cran/causaleffect
+* Date/Publication: 2022-07-14 09:10:05 UTC
+* Number of recursive dependencies: 17
+
+Run `revdepcheck::cloud_details(, "causaleffect")` for more info
+
+</details>
+
+## Newly broken
+
+*   checking examples ... ERROR
+    ```
+    Running examples in ‘causaleffect-Ex.R’ failed
+    The error most likely occurred in:
+    
+    > ### Name: causal.effect
+    > ### Title: Identify a causal effect
+    > ### Aliases: causal.effect
+    > 
+    > ### ** Examples
+    > 
+    > library(igraph)
+    ...
+    > # Here the bidirected edge between X and Z is set to be unobserved in graph g
+    > # This is denoted by giving them a description attribute with the value "U"
+    > # The edges in question are the fourth and the fifth edge
+    > g <- set.edge.attribute(graph = g, name = "description", index = c(4,5), value = "U")
+    Warning: `set.edge.attribute()` was deprecated in igraph 2.0.0.
+    ℹ Please use `set_edge_attr()` instead.
+    > causal.effect("y", "x", G = g)
+    Error in simple_es_index(x, ii) : Unknown edge selected
+    Calls: causal.effect ... [ -> [.igraph.es -> lapply -> FUN -> simple_es_index
+    Execution halted
+    ```
+
+## In both
+
+*   checking re-building of vignette outputs ... WARNING
+    ```
+    Error(s) in re-building vignettes:
+    --- re-building ‘causaleffect.ltx’ using tex
+    Error: processing vignette 'causaleffect.ltx' failed with diagnostics:
+    Running 'texi2dvi' on 'causaleffect.ltx' failed.
+    LaTeX errors:
+    ! LaTeX Error: File `thumbpdf.sty' not found.
+    
+    Type X to quit or <RETURN> to proceed,
+    or enter new name. (Default extension: sty)
+    
+    ...
+    l.16 \usepackage
+                    {csquotes}^^M
+    !  ==> Fatal error occurred, no output PDF file produced!
+    --- failed re-building ‘simplification.ltx’
+    
+    SUMMARY: processing the following files failed:
+      ‘causaleffect.ltx’ ‘simplification.ltx’
+    
+    Error: Vignette re-building failed.
+    Execution halted
+    ```
+
+# centiserve
+
+<details>
+
+* Version: 1.0.0
+* GitHub: NA
+* Source code: https://github.com/cran/centiserve
+* Date/Publication: 2017-07-15 09:34:41 UTC
+* Number of recursive dependencies: 14
+
+Run `revdepcheck::cloud_details(, "centiserve")` for more info
+
+</details>
+
+## Newly broken
+
+*   checking examples ... ERROR
+    ```
+    Running examples in ‘centiserve-Ex.R’ failed
+    The error most likely occurred in:
+    
+    > ### Name: averagedis
+    > ### Title: Find the average distance of a node
+    > ### Aliases: averagedis
+    > 
+    > ### ** Examples
+    > 
+    > g <- graph(c(1,2,2,3,3,4,4,2), directed=FALSE)
+    Warning: `graph()` was deprecated in igraph 2.0.0.
+    ℹ Please use `make_graph()` instead.
+    > averagedis(g)
+    Error in getIgraphOpt("add.vertex.names") && is.named(graph) : 
+      invalid 'x' type in 'x && y'
```

Comment:
We need to inline `igraph_options()` and `igraph_opt()` instead of calling them, due to how the pkgconfig package works.

Reply:
I'm not sure I follow.

Reply:
pkgconfig looks at `sys.frames()` with no way to override.

``` r
pkgconfig:::get_from_session
#> function (key) 
#> {
#>     value <- config[[key]]
#>     if (is.null(value)) 
#>         return(NULL)
#>     pkgs <- sys.frames()
#>     pkgs <- lapply(pkgs, parent.env)
#>     pkgs <- Filter(pkgs, f = isNamespace)
#>     pkgs <- vapply(pkgs, environmentName, "")
#>     pkgs <- unique(pkgs)
#>     for (p in rev(pkgs)) {
#>         if (p %in% names(value)) 
#>             return(value[p])
#>     }
#>     if ("R_GlobalEnv" %in% names(value)) {
#>         return(value["R_GlobalEnv"])
#>     }
#>     NULL
#> }
#> <bytecode: 0x11ac46890>
#> <environment: namespace:pkgconfig>
```

<sup>Created on 2024-01-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

This particular failure occurs after 2074784a0d3d39624ca9c83e764eac91d86302e6.

Reply:
Thanks. I don't understand what you mean by inlining the functions instead of calling them.